#!/usr/bin/env python
PACKAGE = "lane_follow_blob"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


#
# Two lines at angles similar to a road
#

gen.add("l1_x1", int_t,    0, "l1_x1", 188, 0, 400)
gen.add("l1_y1", int_t,    0, "l1_y1", 108, 0, 400)
gen.add("l1_x2", int_t,    0, "l1_x2", 261, 0, 400)
gen.add("l1_y2", int_t,    0, "l1_y2", 320, 0, 400)

gen.add("l2_x1", int_t,    0, "l2_x1", 188, 0, 400)
gen.add("l2_y1", int_t,    0, "l2_y1", 108, 0, 400)
gen.add("l2_x2", int_t,    0, "l2_x2",  24, 0, 400)
gen.add("l2_y2", int_t,    0, "l2_y2", 320, 0, 400)

#
# Two evenly spaced vertical lines
#

# gen.add("l1_x1", int_t,    0, "l1_x1", 100, 0, 400)
# gen.add("l1_y1", int_t,    0, "l1_y1",   0, 0, 400)
# gen.add("l1_x2", int_t,    0, "l1_x2", 100, 0, 400)
# gen.add("l1_y2", int_t,    0, "l1_y2", 400, 0, 400)

# gen.add("l2_x1", int_t,    0, "l2_x1", 300, 0, 400)
# gen.add("l2_y1", int_t,    0, "l2_y1",   0, 0, 400)
# gen.add("l2_x2", int_t,    0, "l2_x2", 300, 0, 400)
# gen.add("l2_y2", int_t,    0, "l2_y2", 400, 0, 400)


exit(gen.generate(PACKAGE, "lane_follow_blob", "Cast"))
